main.nf

```diff
@@ -185,50 +262,15 @@ variant_snp_filter="-fSNP"
 variant_snp_filter=""
 }
 
-if (params.raw_input){
-
-  input_mzmls = Channel.empty()
-  input_mzmls_align = Channel.empty()
-  input_mzmls_unpicked = Channel.empty()
-  input_mzmls_align_unpicked = Channel.empty()
-
-  Channel
-        .fromPath( params.raw_files )
-        .ifEmpty { exit 1, "Cannot find any raw files matching: ${params.raw_files}\nNB: Path needs to be enclosed in quotes!" }
-        .set { input_raws }
-
-
-} else {
-
-  input_raws = Channel.empty()
-
-  if (params.run_centroidisation) {
-    Channel
-        .fromPath( params.mzmls )
-        .ifEmpty { exit 1, "Cannot find any mzmls matching: ${params.mzmls}\nNB: Path needs to be enclosed in quotes!" }
-        .set { input_mzmls_unpicked }
-
-    input_mzmls = Channel.empty()
-    input_mzmls_align = Channel.empty()
-
-  } else {
-    Channel
-        .fromPath( params.mzmls )
-        .ifEmpty { exit 1, "Cannot find any mzmls matching: ${params.mzmls}\nNB: Path needs to be enclosed in quotes!" }
-        .into { input_mzmls; input_mzmls_align }
-
-    input_mzmls_unpicked = Channel.empty()
-    input_mzmls_align_unpicked = Channel.empty()
-  }
-
-}
 
 /*
  * Create a channel for input fasta file
  */
 if( params.include_proteins_from_vcf) {
     Channel
         .fromPath( params.fasta )
+        .spread(ch_samples_for_fasta)
+        .flatMap{it -> [tuple(it[1],it[2],it[0])]}
```

Comment:
sadd comment what it points to

Reply:
Why use `flatMap` + list encapsulation instead of just `map` + tuple?